src/SourceBuild/content/eng/tools/PrBaselinePublisher/Publisher.cs

```diff
@@ -0,0 +1,374 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace PrBaselinePublisher;
+
+using Octokit;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+public class Publisher
+{
+    private string _repoOwner;
+    private string _repoName;
+    private GitHubClient _client;
+
+    public Publisher(string repo, string gitHubToken)
+    {
+        // Create a new GitHub client
+        _client = new GitHubClient(new ProductHeaderValue("PrBaselinePublisher"));
+        var authToken = new Credentials(gitHubToken);
+        _client.Credentials = authToken;
+        _repoOwner = repo.Split('/')[0];
+        _repoName = repo.Split('/')[1];
+    }
+
+    private static readonly string DefaultLicenseBaselineContent = "{\n  \"files\": []\n}";
+    private record ChangedFile(string Path, string Content);
+
+    public async Task<int> ExecuteAsync(
+        string updatedTestsResultsPath,
+        int buildId,
+        string title,
+        string targetBranch,
+        string originalTestResultsPath,
+        Pipelines pipeline)
+    {
+        DateTime startTime = DateTime.Now;
+
+        Log.LogInformation($"Starting PR baseline publisher at {startTime} for pipeline {pipeline}.");
+
+        var updatedTestsFiles = GetUpdatedFiles(updatedTestsResultsPath);
+
+        // Create a new tree for the originalTestResultsPath based on the target branch
+        var testResultsTreeItems = new List<NewTreeItem>();
+        var originalTreeResponse = await _client.Git.Tree.GetRecursive(_repoOwner, _repoName, targetBranch);
+        foreach (var file in originalTreeResponse.Tree)
+        {
+            if (file.Path.Contains(originalTestResultsPath) && file.Path != originalTestResultsPath)
+            {
+                testResultsTreeItems.Add(new NewTreeItem
+                {
+                    Path = Path.GetRelativePath(originalTestResultsPath, file.Path),
+                    Mode = file.Mode,
+                    Type = file.Type.Value,
+                    Sha = file.Sha
+                });
+            }
+        }
```

Comment:
This can be rewritten with LINQ methods.

```suggestion
        var originalTreeResponse = await _client.Git.Tree.GetRecursive(_repoOwner, _repoName, targetBranch);
        var testResultsTreeItems = originalTreeResponse.Tree
            .Where(file => file.Path.Contains(originalTestResultsPath) && file.Path != originalTestResultsPath)
            .Select(file => new NewTreeItem
                {
                    Path = Path.GetRelativePath(originalTestResultsPath, file.Path),
                    Mode = file.Mode,
                    Type = file.Type.Value,
                    Sha = file.Sha
                })
            .ToList();
```